### PR TITLE
feat(client): turn nav-links component functional

### PR DESCRIPTION
**Pull request by ahmaxed**
This pr is a prerequisite for the growthbook AB testing pr as a react hook needs to be called in this component.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: client/src/components/Header/components/nav-links.tsx [hunk 1]
```diff
@@ -54,68 +50,58 @@ const mapDispatchToProps = {
   openSignoutModal
 };
 
-class NavLinks extends Component<NavLinksProps, NavlinkStates> {
-  static displayName: string;
-  langButtonRef: React.RefObject<HTMLButtonElement>;
-  firstLangOptionRef: React.RefObject<HTMLButtonElement>;
-  lastLangOptionRef: React.RefObject<HTMLButtonElement>;
+// type NavlinkProps = <NavLinksProps, NavlinkStates>
```
Comment (ojeytonwilliams) [inline client/src/components/Header/components/nav-links.tsx:53]:
```suggestion
```

File: client/src/components/Header/components/nav-links.tsx [hunk 2]
```diff
@@ -334,247 +313,228 @@ class NavLinks extends Component<NavLinksProps, NavlinkStates> {
...
+            <>
+              <span>{t('settings.labels.night-mode')}</span>
+              {currentUserTheme === Themes.Night ? (
+                <FontAwesomeIcon icon={faCheckSquare} />
+              ) : (
+                <FontAwesomeIcon icon={faSquare} />
+              )}
+            </>
+          ) : (
+            <Fragment key='night-mode'>
```
Comment (ShaunSHamilton) [inline client/src/components/Header/components/nav-links.tsx:445]: Why is this `key` here? What does it do?

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍 

Unrelated to this PR, I just have one question:

**Review by ojeytonwilliams — APPROVED** [review]
The ArrowDown and ArrowUp functions are kind of a mess, but if this is a blocker I can take care of that in another PR.

There's a comment that should be deleted, but I'm happy to merge as is.

**Comment by Sboonny** [comment]
second

File: client/src/components/Header/components/nav-links.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import {
   faExternalLinkAlt
 } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import React, { Component, Fragment, createRef } from 'react';
+import React, { Fragment, RefObject, useRef } from 'react';
```
Comment (Sboonny) [inline client/src/components/Header/components/nav-links.tsx:8]:
```suggestion
import React, { Fragment, useRef } from 'react';
```

File: client/src/components/Header/components/nav-links.tsx [hunk 4]
```diff
@@ -54,68 +50,58 @@ const mapDispatchToProps = {
...
+  toggleNightMode,
+  user,
+  navigate
+}: NavLinksProps) {
+  const langButtonRef =
+    useRef<HTMLButtonElement>() as RefObject<HTMLButtonElement>;
+  const firstLangOptionRef =
+    useRef<HTMLButtonElement>() as RefObject<HTMLButtonElement>;
+  const lastLangOptionRef =
+    useRef<HTMLButtonElement>() as RefObject<HTMLButtonElement>;
```
Comment (Sboonny) [inline client/src/components/Header/components/nav-links.tsx:74]:
```suggestion
  const langButtonRef = useRef<HTMLButtonElement>(null);
  const firstLangOptionRef = useRef<HTMLButtonElement>(null);
  const lastLangOptionRef = useRef<HTMLButtonElement>(null);
```

**Review by Sboonny — CHANGES_REQUESTED** [review]
the `as` provide less secure type, and the errors are displayed because of the lack of initial `null` state

**Comment by Sboonny** [comment]
full free to merge after removing `as`, and if you think `as` are need you can ingnore my request

**Comment by Sboonny** [comment]
ran through Bruce accessibility list and found no issue 👍 

> [ ] When the main navigation menu is expanded and keyboard focus is on an item in main menu, the `Esc` key should close the menu and put keyboard focus on the Menu button.
> 
> [ ] When focus is on the `Change Language` item in the main menu, the up arrow should open the lang menu and place focus on the last item in the lang menu.
> 
> [ ] When focus is on the `Change Language` item in the main menu, the down arrow should open the lang menu and place focus on the first item in the lang menu.
> 
> [ ] When lang menu is opened and keyboard focus is automatically moved into lang menu, the up and down arrow keys should move the keyboard focus up/down one menu item.
> 
> [ ] In the lang menu, when focus is on the last item, arrow down should move focus to first item. Similarly, when focus is on the first item, arrow up should move focus to last item.
> 
> [ ] When focus is on an item in the language menu, `Shift + Tab` should close the lang menu and put focus on the item immediately above the `Change Language` item.
> 
> [ ] When focus is on an item in the language menu, `Tab` should close the lang menu and put focus on item immediately after the `Change Language` item, or if no such item exists then put focus on the next element that can receive focus on the page (which is the Sign In button at this time).
> 
> [ ] When focus is on any item in lang menu, the `Esc` key should close the lang menu and put focus back on the `Change Language` menu item.
> - When keyboard focus is on "Sign out", `Shift+Tab` takes you to "Night Mode", skipping "Change Language" completely.
> - When keyboard focus is on "Sign out", `Esc` moves the focus to "Change Language" instead of closing the menu and moving the focus to the "Menu" button.
> [ ] If keyboard focus is on the last item in the nav menu, using the tab key should move focus to the next active element on the page (this part works) AND the nav menu should close automatically (this part does not work). This bug is only happening when the user is logged in.

File: client/src/components/Header/components/nav-links.tsx (hunk 2 above)
Comment (ahmadabdolsaheb) [reply client/src/components/Header/components/nav-links.tsx:445]: no idea =)

**Review by Sboonny — APPROVED** [review]

File: client/src/components/Header/components/nav-links.tsx (hunk 2 above)
Comment (Sboonny) [reply client/src/components/Header/components/nav-links.tsx:440]: Resolved to merge this, I will open PR removing this, and we can test this there

**Pull request merged**